Sources/AppcuesKit/Presentation/ExperienceRendering/ExperienceStateMachine.swift

```diff
@@ -284,13 +284,28 @@ extension ExperienceStateMachine {
             // be looking for target elements on the view that need to be fully loaded first, and this container
             // may be presenting after a pre-step navigation action that is finalizing the loading of the new view.
             DispatchQueue.main.asyncAfter(deadline: .now() + 0.0) {
-                do {
-                    try package.stepDecoratingTraitUpdater(stepIndex.item, nil)
-                    SdkMetrics.renderStart(experience.requestID)
-                    try package.presenter {
-                        try? machine.transition(.renderStep)
+
+                func presentStep(onError: @escaping (Error) -> Void) {
+                    do {
+                        try package.stepDecoratingTraitUpdater(stepIndex.item, nil)
+                        SdkMetrics.renderStart(experience.requestID)
```

Comment:
ah yes, metrics. I think you are correct. We could move the `renderStart` out of this local function and call it right before we invoke the initial `presentStep` - I think that would give a more accurate picture of what was happening - the delay time would be rolled up into the render time. I'm not sure what is being done with the metrics, honestly, and hopefully either way this would be negligible noise in the larger rolled up metrics.

Reply:
That seems like the best course of action

Reply:
updated in https://github.com/appcues/appcues-ios-sdk/pull/434/commits/63f7193eadee6088e79b8ca161a8e49963a6cbba